engine/crates/wasi-component-loader/src/hooks/authorization.rs

```diff
@@ -0,0 +1,62 @@
+use anyhow::anyhow;
+use grafbase_tracing::span::GRAFBASE_TARGET;
+use wasmtime::{
+    component::{Resource, TypedFunc},
+    Store,
+};
+
+use crate::{names::AUTHORIZATION_HOOK_FUNCTION, state::WasiState, ComponentLoader, ContextMap, ErrorResponse};
+
+pub(crate) type HookParameters = (Resource<ContextMap>, Vec<String>);
+
+pub(crate) type HookResponse = (Result<Vec<Option<ErrorResponse>>, ErrorResponse>,);
```

Comment:
If those are scoped to `hooks::authorization` anyway, might as well drop the `Hook` prefix^^